### DISCUSSION TITLE
Complex Architectural Model: numerical instability error.

### DISCUSSION THREAD

**Discussion by AndreaSabba73** [firemodels/fds#12772] [Q&A] [answered]
Hi, I'm trying to carry out a  test on a building to be used as a library, with complex geometry. I would like to point out that the rather elaborate geometric model was exported from Autodesk Revit by the architectural designer and imported in fbx format through the Pyrosim application.
I performed several tests, always obtaining the interruption of the calculation due to "Numerical Instability", following numerous warnings such as "WARNING: Minimum density, 0.120 kg/m3, clipped in mesh.".
After consulting both the FDS User Guide and the Pyrosim Guide, I tried to repeat the simulation by doing the following:
- I tried using both a single mesh and multiple meshes (to speed up the calculations), without appreciable differences.
 -I changed the dimensions of the meshes (from 0.50x0.50x0.50 up to 0.25x0.25x0.25)
 -I used a fuel with a molecular weight less than 100, chosen from those included in the FDS database
 -I created openings on the external walls, to avoid excessive increase in pressure inside the environment.
- I’ve read the User's Guide section on numerical instabilities.
- I’m using the current FDS release.
 -I have positioned pressure and temperature detectors at all levels, from the analysis of which it is clear that something is not working as it should.
 I have already read previous posts similar to the following on this forum, with no luck.
Despite these changes, I still get a Numerical Instability error, which at this point I believe could be attributable to the complexity of the geometry imported from Revit.
What am I doing wrong?
See attached fds file and output files, I thank in advance anyone who could give me useful suggestions.
[Attached.zip](https://github.com/firemodels/fds/files/14936774/Attached.zip)

**Comment by rmcdermo** [comment]
@AndreaSabba73 May I ask you to please submit this as a [Beta test report](https://github.com/firemodels/fds/issues/new/choose)?  This will help us stay organized and allow to cite you as a beta tester when we release v7.  Thanks

**Reply by AndreaSabba73** [reply]
@rmcdermo As you wrote in the next post, this is not a case of unstructured geometry. Therefore I believe I should not submit it as a Beta Test Report. Do you confirm?
I would rather change the title of the post to avoid further misunderstandings. Thanks

**Comment by nsarifds** [comment]
Your imported geometry is not a solid obstruction, rather a compilation of 0-thickness plates. For example, instead of a solid steel cube you have just the 6 faces a cube would have and air in the middle. Your .out file shows the existence of 3151 pressure zones. I think this should be the main reason you get the density error. 
Dear developers, is the ULMAT solver (User Guide 21.1.1) a solution to these types of situations or is a better input (geometry) needed?

**Comment by rmcdermo** [comment]
OK, I misconstrued "complex geometry" to mean unstructured geometry.  I see not that this is a collection of OBST.

@nsarifds if the OBST are thin, then, yes, using ULMAT may help with pressure zones.  Keep in mind that per mesh you should keep the number of cells under about 50 000.

**Comment by marcosvanella** [comment, marked as answer]
@nsarifds given that the case has a single mesh with 1.3m cells and thousands of pressure zones I expect ULMAT to be very slow. I would suggest in this case to simplify the model, use ULMAT with ~50^3 cell meshes and/or reduce the number of pressure zones using the handy parameter &MISC MINIMUM_ZONE_VOLUME @mcgratta added some time ago.
See the user guide section 10.3.4 Filling Pressure Zones.

**Reply by AndreaSabba73** [reply]
@nsarifds @marcosvanella Thanks for suggestions. I think the simplification of the architectural model is complicated, considering that it's a rather complex model (building on 5 floors, with communications between one floor and another) imported via Pyrosim after fbx export from Revit. Rebuilding it myself from scratch would be very expensive. I will try to understand if there are more suitable export methods (together with BIM Manager), in the meantime I will try to follow the instructions you suggested (ULMAT, number cell meshes, MINIMUM_ZONE_VOLUME). I'll let you know, thank you.

**Reply by rmcdermo** [reply]
Correct---no need for a beta test report here.  My misunderstanding.  No need to change anything.  Thanks
